### PR TITLE
Docs: Add HAProxy rewrite information considering `serve_from_sub_path` setting

### PR DESCRIPTION
**Pull request by simPod**
**What is this feature?**

Updates HAProxy docs

**Why do we need this feature?**

When investigating why bumping SystemJs broke rendering plugin integration I realised I have to use `serve_from_sub_path` set to `true` which requires having no rewrite in proxy server.

**Who is this feature for?**

HAProxy users

**Which issue(s) does this PR fix?**:

Related to https://github.com/grafana/grafana/issues/76180

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by Eve832** [comment]
Added folks from the dev team for review, as I am not qualified.

**Comment by GrafanaWriter** [comment]
@moxious - Can your team take a look and review, please?

**Comment by josmperez** [comment]
@wbrowne - Looking at https://github.com/grafana/grafana/issues/76180, it appears that the issue this doc was intended to resolve was not reproducable and the perosn who reported the issue found a workaround. Can you confirm that we can close this doc issue?

**Comment by simPod** [comment]
@josmperez read the issue more carefully please.

If you use serve from subpath, you should not handle serving from subpath in your proxy.

In the issue I have removed subpath rewrite from my nginx. I also did the same with my grafana installations where I use haproxy and I have reflected those changes in the PR.

Compare it to nginx docs where the case is specified. It is missing in haproxy docs now.

**Comment by wbrowne** [comment]
> @josmperez read the issue more carefully please.
> 
> If you use serve from subpath, you should not handle serving from subpath in your proxy.
> 
> In the issue I have removed subpath rewrite from my nginx. I also did the same with my grafana installations where I use haproxy and I have reflected those changes in the PR.
> 
> Compare it to nginx docs where the case is specified. It is missing in haproxy docs now.

I think this makes sense to do 👍 It may also be worth adding a note of:

>"If you use serve from subpath, you should not handle serving from subpath in your proxy."

somewhere at the top where it specifies the config sample. WDYT?

**Comment by wbrowne** [comment]
Sorry I more meant at the top of the doc @simPod. @josmperez and I can work together to figure out what makes sense there and get this in a mergeable state.

**Comment by simPod** [comment]
I've added something to start with. Waiting for input.

**Review by wbrowne — APPROVED** [review]
LGTM. Will let @josmperez have the final look.

Edit: Or @grafana/backend-platform who I believe would be the SMEs here

File: docs/sources/tutorials/run-grafana-behind-a-proxy/index.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ You can also serve Grafana behind a _sub path_, such as `http://example.com/graf
 To serve Grafana behind a sub path:
 
 - Include the sub path at the end of the `root_url`.
```
Comment (josmperez) [inline docs/sources/tutorials/run-grafana-behind-a-proxy/index.md:41]:
```suggestion
1. Include the sub path at the end of the `root_url`.
```

File: docs/sources/tutorials/run-grafana-behind-a-proxy/index.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ You can also serve Grafana behind a _sub path_, such as `http://example.com/graf
...
-- Set `serve_from_sub_path` to `true`.
+- Set `serve_from_sub_path` to `true`, **OR** let proxy rewrite the path for you (see proxy config examples bellow).
```
Comment (josmperez) [inline docs/sources/tutorials/run-grafana-behind-a-proxy/index.md:42]:
```suggestion
1. Set `serve_from_sub_path` to `true`. Or, let proxy rewrite the path for you (refer to examples bellow).
```

File: docs/sources/tutorials/run-grafana-behind-a-proxy/index.md [hunk 3]
```diff
@@ -149,11 +149,18 @@ frontend http-in
   use_backend grafana_backend if { path /grafana } or { path_beg /grafana/ }
 
 backend grafana_backend
-  # Requires haproxy >= 1.6
-  http-request set-path %[path,regsub(^/grafana/?,/)]
+  server grafana localhost:3000
+```
+
+If your Grafana configuration does not set `server.serve_from_sub_path` to `true` then you need to add a rewrite rule to `backend grafana_backend` block:
```
Comment (josmperez) [inline docs/sources/tutorials/run-grafana-behind-a-proxy/index.md:155]:
```suggestion
If your Grafana configuration doen't set `server.serve_from_sub_path` to `true`, then you must add a rewrite rule to the `backend grafana_backend` block:
```

**Review by josmperez — APPROVED** [review]
Requires a few stylistic changes, but otherwise LGTM in regard to Grafana style. I haven't confirmed the technical accuracy of this PR.

File: docs/sources/tutorials/run-grafana-behind-a-proxy/index.md [hunk 4]
```diff
@@ -149,11 +149,18 @@ frontend http-in
   use_backend grafana_backend if { path /grafana } or { path_beg /grafana/ }
 
 backend grafana_backend
-  # Requires haproxy >= 1.6
-  http-request set-path %[path,regsub(^/grafana/?,/)]
+  server grafana localhost:3000
+```
+
+If your Grafana configuration doen't set `server.serve_from_sub_path` to `true`, then you must add a rewrite rule to the `backend grafana_backend` block:
```
Comment (simPod) [inline docs/sources/tutorials/run-grafana-behind-a-proxy/index.md:155]:
```suggestion
If your Grafana configuration doesn't set `server.serve_from_sub_path` to `true`, then you must add a rewrite rule to the `backend grafana_backend` block:
```

File: docs/sources/tutorials/run-grafana-behind-a-proxy/index.md [hunk 5]
```diff
@@ -38,8 +38,8 @@ You can also serve Grafana behind a _sub path_, such as `http://example.com/graf
 
 To serve Grafana behind a sub path:
 
-- Include the sub path at the end of the `root_url`.
-- Set `serve_from_sub_path` to `true`.
+1. Include the sub path at the end of the `root_url`.
+1. Set `serve_from_sub_path` to `true`. Or, let proxy rewrite the path for you (refer to examples bellow).
```
Comment (wbrowne) [inline docs/sources/tutorials/run-grafana-behind-a-proxy/index.md:42]:
```suggestion
1. Set `serve_from_sub_path` to `true`. Or, let proxy rewrite the path for you (refer to examples below).
```

**Review by wbrowne — CHANGES_REQUESTED** [review]
Just found a typo. Other than that, LGTM!

**Review by wbrowne — APPROVED** [review]

**Pull request merged**